carcara/src/parser/mod.rs

```diff
@@ -1222,6 +1319,89 @@ impl<'a, R: BufRead> Parser<'a, R> {
         Ok(inner)
     }
 
+    fn parse_indexed_operator(&mut self) -> CarcaraResult<(IndexedOperator, Vec<Constant>)> {
+        let bv_symbol = self.expect_symbol()?;
+        if let Some(value) = bv_symbol.strip_prefix("bv") {
+            let parsed_value = value.parse::<Integer>().unwrap();
+            let mut args = self.parse_sequence(Self::parse_constant, true)?;
+            args.insert(0, Constant::Integer(parsed_value));
+            return Ok((IndexedOperator::BvConst, args));
+        }
+        let op = IndexedOperator::from_str(bv_symbol.as_str()).unwrap();
+        let args = self.parse_sequence(Self::parse_constant, true)?;
+        Ok((op, args))
+    }
+
+    /// Constructs, check operation arguments and sort checks an indexed operation term.
+    fn make_indexed_op(
+        &mut self,
+        op: IndexedOperator,
+        op_args: Vec<Constant>,
+        args: Vec<Rc<Term>>,
+    ) -> Result<Rc<Term>, ParserError> {
+        let sorts: Vec<_> = args.iter().map(|t| self.pool.sort(t)).collect();
+        match &op {
+            IndexedOperator::BvConst => {
+                assert_num_args(&op_args, 2)?;
+                assert_num_args(&args, 0)?;
+                let Constant::Integer(value) = op_args[0].clone() else {
+                    return Err(ParserError::ExpectedIntegerConstant(op_args[0].clone()));
+                };
+                let Constant::Integer(width) = op_args[1].clone() else {
+                    return Err(ParserError::ExpectedIntegerConstant(op_args[1].clone()));
+                };
+                assert_indexed_op_args_value(&[op_args[0].clone()], 0..)?;
+                assert_indexed_op_args_value(&[op_args[1].clone()], 1..)?;
```

Comment:
had to update this to allow constants like `(_ bv0 4)` to be correctly parsed.
@bpandreotti